src/main/java/brooklyn/entity/proxy/aws/ElbControllerImpl.java

```diff
@@ -579,15 +689,101 @@ protected void deleteLoadBalancer(String elbName) {
+                "displayName", "Delete security groups",
+                "description", "Deleting " + removals,
+                "tag", BrooklynTaskTags.tagForCallerEntity(this)
+            ), removals);
+            TaskTags.markInessential(invoke);
+            DynamicTasks.queueIfPossible(invoke).orSubmitAsync(this).andWaitForSuccess();
+        }
+    }
+
+    private Task<Boolean> removeOneSecurityGroup(final AmazonEC2Client client, final String groupId) {
+        return Tasks.create(groupId, new Callable<Boolean>() {
+
+            @Override
+            public Boolean call() throws Exception {
+                ReferenceWithError<Boolean> result;
+                final Repeater repeater = Repeater.create("Delete " + groupId)
+                    .limitTimeTo(2L, TimeUnit.MINUTES)
+                    .every(10, TimeUnit.SECONDS)
+                    .until(new DeleteASecurityGroup(client, groupId))
+                    .rethrowException();
+
+                Tasks.setBlockingDetails(repeater.getDescription());
+                try {
+                    result = repeater.runKeepingError();
+                } finally {
+                    Tasks.resetBlockingDetails();
+                }
+
+                if (Boolean.TRUE.equals(result.getWithoutError()))
+                    return true;
+                if (result.hasError())
+                    throw Exceptions.propagate(result.getError());
+                return false;
+            }
+        });
+    }
+
+    private static class DeleteASecurityGroup implements Callable<Boolean> {
+        private AmazonEC2Client client;
+        private String groupId;
+        public DeleteASecurityGroup(final AmazonEC2Client client, final String groupId) {
+            this.client = client;
+            this.groupId = groupId;
+        }
+
+        /**
+         * Delete the group. If it is still in use, return false to indicate that a retry should be attempted.
+         * @return true unless a retry should be attempted because the group is still in use.
+         */
+        @Override
+        public Boolean call() throws Exception {
+            final DeleteSecurityGroupRequest deleteGroup = new DeleteSecurityGroupRequest().withGroupId(groupId);
+            boolean success = true;
+            try {
+                LOG.debug("Removing security group {}", groupId);
+                client.deleteSecurityGroup(deleteGroup);
+                LOG.debug("Removed security group {}", groupId);
+            } catch (AmazonServiceException ase) {
+                Exceptions.propagateIfFatal(ase);
+                if (ase.getErrorCode().equals(ERROR_CODE_FOR_GROUP_IN_USE)) {
+                    LOG.debug("Security group in use, not deleted {} ", groupId);
+                    success = false;
+                } else {
+                    // TODO For review, should we just propagate this?
+                    LOG.warn("Unexpected exception attempting to delete security group " + groupId, ase);
```

Comment:
I think logging rather than propagating is ok. Does doing so at `warn` risk spamming logs? Especially when an exception is included. The pattern elsewhere in Brooklyn has been to log a line at info/above and the whole exception at debug.